### PR TITLE
Packet Broker network listing and filtering

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

References #3771 

#### Changes

- Update Packet Broker API
- Add search flags for networks
- List all networks, optionally with or without routing policies

#### Testing

Unit tests and testing with locally running PB

##### Regressions

None

#### Notes for Reviewers

@htdvisser is the main reviewer

@kschiffer please use this as base for implementing #3771.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by adriansmares — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -108,10 +108,10 @@ require (
 	github.com/valyala/fasttemplate v1.1.0 // indirect
 	github.com/vmihailenco/msgpack/v5 v5.0.0-beta.1
 	go.opencensus.io v0.22.3
-	go.packetbroker.org/api/iam v1.2.2-tts
-	go.packetbroker.org/api/iam/v2 v2.1.2-tts
-	go.packetbroker.org/api/routing v1.2.2-tts
-	go.packetbroker.org/api/v3 v3.4.2-tts
+	go.packetbroker.org/api/iam v1.4.2-tts
```
Comment (adriansmares) [inline go.mod:111]: Needs `go mod tidy` in `tools`.

Comment (adriansmares) [inline go.mod:122]: Needs `go mod tidy` in `tools`.

Comment (johanstokking) [reply go.mod:122]: It needed a `go mod download` in root and in `tools` ....

**Review by htdvisser — APPROVED** [review]
Nothing to add

**Pull request merged**